From 2a6ed25a8e3276b7a483f4df31af980b48527eb9 Mon Sep 17 00:00:00 2001
From: Ryoichi Yasutake <ryoichi.x.yasutake@sonymobile.com>
Date: Wed, 21 Feb 2018 20:53:25 +0900
Subject: [PATCH 12/12] Request to reset effect buffer in clearInputBuffer

When switching track, previous data may remain in work buffer of
effect, and causes pop noise at the begining of next track.
So send reset request to effect during switching track
in order to clear useless data.

Bug: 73720726
Test: No pop noise with AudioEffect when switching track
Change-Id: Iaeb4ab928303310427032617e9398b08f4abe5fa
---
 services/audioflinger/Effects.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/audioflinger/Effects.cpp b/services/audioflinger/Effects.cpp
index 2047dfd2a..5840b03a1 100644
--- a/services/audioflinger/Effects.cpp
+++ b/services/audioflinger/Effects.cpp
@@ -1900,6 +1900,10 @@ void AudioFlinger::EffectChain::clearInputBuffer()
         return;
     }
     clearInputBuffer_l(thread);
+
+    for (size_t i = 0; i < mEffects.size(); i++) {
+        mEffects[i]->reset_l();
+    }
 }
 
 // Must be called with EffectChain::mLock locked
-- 
2.17.1

